lib/karafka/callbacks/dsl.rb

```diff
@@ -1,14 +1,29 @@
 # frozen_string_literal: true
 
 module Karafka
+  # Callbacks. This module is deprecated.
   module Callbacks
     # App level dsl to define callbacks
     module Dsl
-      Callbacks::TYPES.each do |callback_type|
-        # Allows us to define a block, that will be executed for a given moment
-        # @param [Block] block that should be executed after the initialization process
-        define_method callback_type do |&block|
-          config.callbacks.send(callback_type).push block
+      # Allows us to define a block that will be executed after Karafka has been initialized
+      # @yield [config] block that should be executed after the initialization process
+      # @yieldparam [Hash] config Karafka config
+      def after_init
+        Karafka.event_publisher.subscribe('app.after_init') do |payload|
+          # We have to unpack the payload to keep backward-compatibility
+          yield payload[:config]
```

Comment:
we don;t need backwards compatibility outside of the  consumer callbacks. The "non" consumer callbacks backwards compatibility can be dropped :dancer:

Reply:
Are you sure? This can break someone's app on boot time. 

If we want to drop backwards-compatibility here, we could get rid of `Karafka::Callbacks` and let users subscribe to `app.after_init` and `connection.listener.before_fetch_loop` by themselves:

```ruby
# config/initializers/foo.rb
Karafka.event_publisher.subscribe('app.after_init') do |payload|
  Foo.initialize
end

# karafka.rb
require_relative 'config/initializers/foo'

class App < Karafka::App
 # ...
end

App.boot!
```

Reply:
I am. This is targeted for the next bigger release (1.3), so there will be release notes and migration details :)